lua/compe/completion/init.lua

```diff
@@ -39,14 +36,21 @@ function Completion.unregister_source(self, id)
   end
 end
 
---- on_insert_leave
-function Completion.on_insert_leave(self)
+--- on leaving insert
+-- internal private function.
+-- receives vim's InsertLeave autocmd
+-- @param self
+-- @function on_insert_leave
+M.on_insert_leave = function(self)
   self:clear()
-  VimBridge.clear()
+  require'compe.completion.source.vim_bridge'.clear()
```

Comment:
I think we should put `require` for the top of file. What do you think about it?

Reply:
@hrsh7th Yes, `require` should be on the top of the file only when it is used more then once or more then two functions.
Otherwise, its waste on resources to `require` module on the top of the file and just use once and it makes top file less populated with defintion of locals.  What do you think?

Reply:
I love consistency rather than efficiency.

I don't want to worry if the module is being used elsewhere...

Reply:
@hrsh7th  You know what I totally agree with you on that. But be aware of those local require as it may impact performance if the table is huge or does side effects when required. 

Should I push another commit to fix this, or you will do it when merging? thanks

Reply:
OK. I can fix it after merge this. Thanks!